app-modules/assistant/src/Listeners/LogAssistantChatMessage.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Assist\Assistant\Listeners;
+
+use Illuminate\Support\Arr;
+use Assist\IntegrationAI\Events\AIPromptInitiated;
+
+class LogAssistantChatMessage
+{
+    public function handle(AIPromptInitiated $event): void
+    {
+        $prompt = $event->prompt;
+
+        $prompt->user->assistantChatMessageLogs()->create([
+            'message' => $prompt->message,
+            'metadata' => $prompt->metadata,
+            'request' => [
+                'ip' => $prompt->request->ip(),
+                'headers' => Arr::except(
+                    $prompt->request->headers->all(),
+                    ['cookie'],
+                ),
+            ],
+        ]);
```

Comment:
@danharrin What are your thoughts on adding the `timestamp` from the DTO here? 

My initial thinking was that this would be a queued process, so the creation time for this might not be a reliable indicator of when the message was actually sent.

Reply:
I agree with this.

Reply:
Done